### PR TITLE
DEP: Assigning CRS to a GeoDataFrame without a geometry column

### PR DESCRIPTION
**Pull request by m-richards**
In the new error messages, I've tried to give some guidance about how to fix the errors if people run into them - I'm not sure if this is helpful or overkill - particularly for the `__init__` where refering a user to the documentation might be more helpful.

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
> I'm not sure if this is helpful or overkill 

That's certainly not overkill, clear and informative error message are really helpful for users.

File: geopandas/tests/test_crs.py [hunk 1]
```diff
@@ -340,13 +338,9 @@ def test_dataframe_setitem(self):
         "scalar", [None, Point(0, 0), LineString([(0, 0), (1, 1)])]
     )
     def test_scalar(self, scalar):
-        with pytest.warns(FutureWarning):
+        with pytest.raises(ValueError, match="Assigning CRS to a GeoDataFrame without"):
             df = GeoDataFrame()
```
Comment (jorisvandenbossche) [inline geopandas/tests/test_crs.py:342]: Can you update this test to use a valid GeoDataFrame, and add a separate test for asserting the error? (checking that setting the crs with a scalar number works is still useful)

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good!

**Comment by martinfleis** [comment]
Once this is merged (I have no other comments apart from those mentioned by @jorisvandenbossche) I can try to eliminate `_crs` from the GeoDataFrame as discussed yesterday. I had a quick look and it may not be super straightforward but it should be possible.

**Comment by jorisvandenbossche** [comment]
See also the last bullet point in https://github.com/geopandas/geopandas/issues/2133#issuecomment-1024111760, we might want to deprecate it first to access the CRS when there is no active geometry column (although this is certainly for corner cases, especially after this PR).

**Review by martinfleis — APPROVED** [review]

**Comment by martinfleis** [comment]
Thanks @m-richards!

**Pull request merged**